pkg/operator/resource/resourceapply/admissionregistration.go

```diff
@@ -0,0 +1,185 @@
+// it will be created. If it does exist, the metadata of the required
+// mutatingwebhookconfiguration will be merged with the existing mutatingwebhookconfiguration
+// and an update performed if the mutatingwebhookconfiguration spec and metadata differ from
+// the previously required spec and metadata. For further detail, check the top-level comment.
+func ApplyMutatingWebhookConfiguration(client admissionregistrationclientv1.MutatingWebhookConfigurationInterface, recorder events.Recorder,
+	requiredOriginal *admissionregistrationv1.MutatingWebhookConfiguration, expectedGeneration int64) (*admissionregistrationv1.MutatingWebhookConfiguration, bool, error) {
+
+	required := requiredOriginal.DeepCopy()
+	err := SetSpecHashAnnotation(&required.ObjectMeta, required.Webhooks)
+	if err != nil {
+		return nil, false, err
+	}
+
+	existing, err := client.Get(context.TODO(), required.GetName(), metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		created, err := client.Create(context.TODO(), required, metav1.CreateOptions{})
+		reportCreateEvent(recorder, created, err)
+		return created, true, err
+	} else if err != nil {
+		return nil, false, err
+	}
+
+	// Providing upgrade compatibility with service-ca-bundle operator
+	// and ignore clientConfig.caBundle changes on "inject-cabundle" annotation
+	if required.GetAnnotations()[genericCABundleInjectorAnnotation] != "" {
+		if err := copyMutatingWebhookCABundle(required, *existing); err != nil {
+			return nil, false, err
+		}
+	}
+
+	toWrite := existing.DeepCopy()
+	modified := pointer.BoolPtr(false)
+	resourcemerge.EnsureObjectMeta(modified, &toWrite.ObjectMeta, required.ObjectMeta)
+	// there was no change to metadata, the generation was right
+	if !*modified && toWrite.ObjectMeta.Generation == expectedGeneration {
+		return existing, false, nil
+	}
+	toWrite.Webhooks = required.Webhooks
+
+	if klog.V(4) {
+		klog.Infof("MutatingWebhookConfiguration %q changes: %v", required.GetNamespace()+"/"+required.GetName(), JSONPatchNoError(existing, required))
+	}
+
+	actual, err := client.Update(context.TODO(), toWrite, metav1.UpdateOptions{})
+	reportUpdateEvent(recorder, actual, err)
+	return actual, true, err
+}
+
+func copyMutatingWebhookCABundle(required *admissionregistrationv1.MutatingWebhookConfiguration,
+	existing admissionregistrationv1.MutatingWebhookConfiguration) error {
+	existingMutatingWebhooks := make(map[string]admissionregistrationv1.MutatingWebhook)
+	for _, mutatingWebhook := range existing.Webhooks {
+		existingMutatingWebhooks[mutatingWebhook.Name] = mutatingWebhook
+	}
+
+	webhooks := make([]admissionregistrationv1.MutatingWebhook, len(required.Webhooks))
+	for i, mutatingWebhook := range required.Webhooks {
+		if webhook, ok := existingMutatingWebhooks[mutatingWebhook.Name]; ok {
+			mutatingWebhook.ClientConfig.CABundle = webhook.ClientConfig.CABundle
+		}
+		webhooks[i] = mutatingWebhook
+	}
+	required.Webhooks = webhooks
```

Comment:
note that Webhooks here is the same type as for validating webhook. You can easily merge this func with copyValidatingWebhookCABundle by using another function type.